### PR TITLE
[Documentation] Add documentations for non-packed tensors convolution

### PR DESCRIPTION
**Pull request by iq136boy**

### CODE REVIEW

**Comment by junliume** [comment]
@iq136boy do you also need to update here? https://github.com/ROCmSoftwarePlatform/MIOpen/tree/develop/docs

**Comment by iq136boy** [comment]
> @iq136boy do you also need to update here? https://github.com/ROCmSoftwarePlatform/MIOpen/tree/develop/docs

@junliume Maybe not. I did not add new APIs for convolution. My understanding is that the docs about convolution need to be edited only if a new API is added.

**Review by junliume — APPROVED** [review]

File: include/miopen/miopen.h [hunk 1]
```diff
@@ -1730,6 +1730,14 @@ miopenFindConvolutionForwardAlgorithm(miopenHandle_t handle,
  * The scaling parameter alpha (float) and shift parameter beta (float) are only supported for
```
Comment (atamazov) [inline include/miopen/miopen.h:1733]: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/8be2998d7a1210b7bc82d778f7df0d92aa65b57d/include/miopen/miopen.h#L636

Mention that it creates _packed_ tensor with NCHW layout.

File: include/miopen/miopen.h [hunk 2]
```diff
@@ -1730,6 +1730,14 @@ miopenFindConvolutionForwardAlgorithm(miopenHandle_t handle,
...
  * alpha = 1 and beta = 0.
```
Comment (atamazov) [inline include/miopen/miopen.h:1734]: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/8be2998d7a1210b7bc82d778f7df0d92aa65b57d/include/miopen/miopen.h#L667

Here mention that is allows creation of unpacked tensor.

Maybe this is a good place to give a definition of unpacked tensor.

File: include/miopen/miopen.h [hunk 3]
```diff
@@ -1730,6 +1730,14 @@ miopenFindConvolutionForwardAlgorithm(miopenHandle_t handle,
...
  *
+ * The forward convolution is designed to accommodate both packed and non-packed tensor strides for
+ * multiple data types and dimensions across various platforms. This flexibility ensures optimal
+ * performance in handling diverse computational scenarios. To configure tensor parameters,
+ * including strides, users can utilize the APIs miopenSetTensorDescriptor() and
+ * miopenGetTensorDescriptor(). These APIs empower developers to seamlessly set and retrieve tensor
+ * information, facilitating a more intuitive and efficient workflow. The tensor strides are
+ * non-packed by default.
+ *
  * If using Group/Depthwise convolution mode, call miopenSetConvolutionGroupCount() before running
```
Comment (atamazov) [inline include/miopen/miopen.h:1744]: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/8be2998d7a1210b7bc82d778f7df0d92aa65b57d/include/miopen/miopen.h#L651

Interface for setting N-D _packed_ tensor shape

File: include/miopen/miopen.h [hunk 4]
```diff
@@ -1730,6 +1730,14 @@ miopenFindConvolutionForwardAlgorithm(miopenHandle_t handle,
...
  * this.
```
Comment (atamazov) [inline include/miopen/miopen.h:1745]: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/8be2998d7a1210b7bc82d778f7df0d92aa65b57d/include/miopen/miopen.h#L721-L722

This sounds strange: "MIOpen has support for 1, 2, 3, 4, 5 dimensional tensor of layout"

I think it is worth mentioning unpacked tensors here.

Comment (Rmalavally) [reply include/miopen/miopen.h:1745]: @junliume @iq136boy is the documentation final?

Comment (iq136boy) [reply include/miopen/miopen.h:1745]: > @junliume @iq136boy is the documentation final?

@Rmalavally @junliume The final version unless there are more comments.

Comment (Rmalavally) [reply include/miopen/miopen.h:1745]: Thank you for confirming. Much appreciated. I will now review the content and recommend changes as needed.

File: include/miopen/miopen.h [hunk 5]
```diff
@@ -634,7 +634,7 @@ MIOPEN_EXPORT miopenStatus_t miopenCreateTensorDescriptor(miopenTensorDescriptor
 
 /*! @brief Set shape of 4D tensor
  *
- * Interface for setting 4-D packed tensor shape.
+ * Interface for setting 4-D tensor shape. MIOpen currently implements NCHW and NHWC layout.
```
Comment (Rmalavally) [inline include/miopen/miopen.h:637]: Could we please change the sentence to,

MIOpen currently implements both NHWC (Number of samples, Height, Channels, Width) and NCHW (Number of samples, Channels, Height, Width) layouts. 

@junliume 

Let me know if I have misunderstood the abbreviations. Thanks so much.

Comment (junliume) [reply include/miopen/miopen.h:637]: @iq136boy could you follow up with the comment?

Comment (atamazov) [reply include/miopen/miopen.h:637]: @Rmalavally "Number of samples" (N) is actually "Batch size"

File: include/miopen/miopen.h (hunk 4 above)
Comment (atamazov) [reply include/miopen/miopen.h:1745]: [resolved]

**Review by atamazov — APPROVED** [review]
🟢 LGTM

**Comment by Rmalavally** [comment]
> 🟢 LGTM

Thanks so much.

**Review by Rmalavally — APPROVED** [review]

**Review by Rmalavally — APPROVED** [review]
Thanks, Artem and Jun, for approving the changes.

**Pull request merged**